bee-protocol/src/worker/milestone_cone_updater.rs

```diff
@@ -74,21 +74,22 @@ async fn update_messages_referenced_by_milestone<N: Node>(
 ) where
     N::Backend: StorageBackend,
 {
-    let mut to_visit = vec![message_id];
-    let mut visited = HashSet::new();
+    let mut parents_to_visit = vec![message_id];
+    let mut parents_visited = HashSet::new();
+    let mut children_to_visit = HashSet::new();
 
-    while let Some(ref message_id) = to_visit.pop() {
-        if visited.contains(message_id) {
+    while let Some(ref message_id) = parents_to_visit.pop() {
+        if parents_visited.contains(message_id) {
             continue;
         } else {
-            visited.insert(*message_id);
+            parents_visited.insert(*message_id);
         }
 
         if tangle.is_solid_entry_point(&message_id) {
             continue;
         }
 
-        // maybe the check below is not necessary; all messages from the most recent cone should be present
+        // maybe this check is not necessary; all messages from the most recent cone should be present
         if let Some((parent1, parent2)) = tangle.get(&message_id).await.map(|m| (*m.parent1(), *m.parent2())) {
             // unwrap() is safe since message is present and so is the metadata
             if tangle.get_metadata(&message_id).await.unwrap().cone_index().is_some() {
```

Comment:
so this whole algorithm starts with the milestone itself, right? Shouldn't we start with the parents of that milestone? Or is this also the first time, where we set that metadata for a milestone?

Reply:
forget that :smile: